/*
 * Copyright 2024 The PhoenixOS Authors. All rights reserved.
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package pos_protobuf;

/*!
 *  \brief  binary format of POSHandle
 */
message Bin_POSHandle {
    uint64 id = 1;
    uint32 resource_type_id = 2;
    uint64 client_addr = 3;
    uint64 server_addr = 4;
    uint64 size = 5;
    repeated uint64 parent_handle_resource_type_idx = 6;
    repeated uint64 parent_handle_idx = 7;
    uint64 state_size = 8;
    uint32 state_type = 9;
    bytes state = 10;
}
